### PR TITLE
adding support to view file from stacktraces

### PR DESCRIPTION
**Pull request by miki725**
Added a way to view files from stacktrace. Useful when looking at SQL panel which allows to view code directly in debugging without having a need to jump to editor.

image

image

### CODE REVIEW

File: debug_toolbar/static/debug_toolbar/css/toolbar.css [hunk 1]
```diff
@@ -615,16 +615,35 @@
     margin-bottom: 3px;
     font-family:Consolas, Monaco, "Bitstream Vera Sans Mono", "Lucida Console", monospace;
 }
+#djDebug .djdt-stack {
+	white-space: normal;
+}
 #djDebug .djdt-stack span {
-    color: #000;
```
Comment (miki725) [inline debug_toolbar/static/debug_toolbar/css/toolbar.css]: this file is inconsistent with indents. some are tabs and some are spaces. let me know if I should make it consistent in whole file.

File: debug_toolbar/static/debug_toolbar/js/toolbar.js [hunk 2]
```diff
@@ -111,20 +111,16 @@
                 var name = this.tagName.toLowerCase();
                 var ajax_data = {};
 
-                if (name == 'button') {
+                if (name === 'button' || name === 'a') {
                     var form = this.closest('form');
-                    ajax_data.url = this.getAttribute('formaction');
+                    ajax_data.url = name === 'button' ? this.getAttribute('formaction') : this.getAttribute('href');
```
Comment (miki725) [inline debug_toolbar/static/debug_toolbar/js/toolbar.js]: not sure if this has side-effects anywhere else. I could switch to using a button to view file however that did not looked well so switched back to using a link

**Comment by matthiask** [comment]
The templates panel already does a similar thing, without a form though but with django.core.signing:
https://github.com/jazzband/django-debug-toolbar/blob/master/debug_toolbar/panels/templates/views.py#L14

It would be great to reuse the implementation we already have if possible.

**Comment by miki725** [comment]
sure. I can take a look. didnt realize template already did something similar. I see it also uses pygments so Ill try to include that as well

**Comment by matthiask** [comment]
I'm closing this pull request because of inactivity, not rejecting the feature. Please resubmit if you still want to work on this, after resolving conflicts and adapting the code for current versions of Django etc.

**Pull request closed without merging**